src/Unosquare.Sparkfun.FingerprintModule/FingerprintReader.cs

```diff
@@ -507,26 +534,25 @@ public Task<BasicResponse> SetSecurityLevelAsync(int level, CancellationToken ct
         /// <summary>
         /// Opens and initialize the fingerprint device at the specified port name with the specified baud rate.
         /// </summary>
-        /// <param name="portName">Name of the port.</param>
+        /// <param name="serialPort">The serial port.</param>
         /// <param name="baudRate">The baud rate.</param>
-        /// <param name="ct">An instance of <see cref="CancellationToken"/></param>
-        /// <returns>A task that represents the asynchronous open operation.</returns>
+        /// <param name="skipBaudRate">if set to <c>true</c> [skip baud rate].</param>
+        /// <param name="ct">An instance of <see cref="CancellationToken" />.</param>
+        /// <returns>
+        /// A task that represents the asynchronous open operation.
+        /// </returns>
         /// <exception cref="Exception">The device could not be initialized.</exception>
-        private async Task OpenAsync(string portName, int baudRate, CancellationToken ct)
+        private async Task OpenAsync(ISerialPort serialPort, int baudRate, bool skipBaudRate, CancellationToken ct)
         {
-            _serialPort =
-#if NET452
-                new MsSerialPort(portName, baudRate);
-#else
-                new RjcpSerialPort(portName, baudRate);
-#endif
+            _serialPort = serialPort;
+
             _serialPort.Open();
             await Task.Delay(100, ct);
 
-            if (baudRate != TargetBaudRate)
+            if (!skipBaudRate || baudRate != TargetBaudRate)
```

Comment:
if *skipBaudRate* is **false** this condition always will be true no matter the value of the baud rate.